### PR TITLE
add tutorial on best practices as preparation for testing tutorial

### PR DESCRIPTION
**Pull request by nilsnolde**
What the title says.. A little dry, no new functionality, but more a best practice guide & some useful tips. Maybe you have some insight for MacOS on extending PYTHONPATH so an IDE can recognize qgis symbols.

### CODE REVIEW

File: qgis/QGIS_PluginProperWay.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+## 2 - Enable IntelliSense
+
+Let's continue with the most gratifying and useful tip: enable your IDE to provide you with intelligent code completion. This will save you tons of time grooming through QGIS and Qt documentation. While QGIS [docstrings](https://en.wikipedia.org/wiki/Docstring#:~:text=In%20programming%2C%20a%20docstring%20is,a%20specific%20segment%20of%20code.) are outstanding, PyQt pretty much has zero docstrings though (but you still see which functions are even available for a module/class).
+
+The exact method depends a lot on the IDE and platform (Linux, MacOS, Win), but the general idea is always the same: you have to tell your IDE where the QGIS packages are located. The process of doing that obviously also depends on your IDE; for PyCharm see [this SO thread](https://stackoverflow.com/questions/48947494/add-directory-to-python-path-in-pycharm).
+
+To really understand what's goind on, there's a couple of gotcha's in the way QGIS and Python interact:
```
Comment (TimMcCauley) [inline qgis/QGIS_PluginProperWay.md:68]: "goind" typo

File: qgis/QGIS_PluginProperWay.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+The code inside the plugin package is commonly leaning towards QGIS own logic:
+- `core/`: Contains the business logic of your plugin, mostly functionality that doesn't depend on the rest of your plugin like the UI. In our case that'd would the logic to request Nominatim and the map tool capturing clicked points.
+- `gui`: Contains the logic around your dialogs, which will make use of the `core/` functionality.
+- `icons`: A place to hold your plugin-specific icons, doesn't need to be a Python package (i.e. no need for a `__init__.py`).
+- `ui`: Contains the raw and/or compiled UI files.
+- `processing` (if needed): Contains the modules used to integrate with QGIS Processing framework.
+- `__init__.py`: **Only** contains the `classFactory()` QGIS expects every plugin to have.
+- `quick_api.py`: The class QGIS uses to instantiate the plugin. This should only contain functionality which is important to the **main** QGIS app, such as adding toolbars/menus (e.g. in `initGui()`) or the `unload()` function. See our [QGIS Plugin 101 tutorial](https://gis-ops.com/qgis-3-plugin-tutorial-plugin-development-reference-guide/) for more info.
+
+## 6 - Linters
```
Comment (TimMcCauley) [inline qgis/QGIS_PluginProperWay.md:159]: Would also be nice to add a code formatter which runs before the linter, i.e. black, I think it works quite well together with flake8, https://ljvmiranda921.github.io/notebook/2018/06/21/precommits-using-black-and-flake8/

File: qgis/QGIS_PluginProperWay.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+- [QGIS 3 Plugins - Signals and Slots in PyQt](https://gis-ops.com/qgis-3-plugin-tutorial-pyqt-signal-slot-explained/)
+- [QGIS 3 Plugins - Plugin Development Part 1](https://gis-ops.com/qgis-3-plugin-tutorial-plugin-development-explained-part-1/)
+- [QGIS 3 Plugins - Plugin Development Part 2](https://gis-ops.com/qgis-3-plugin-tutorial-plugin-development-explained-part-2/)
+- [QGIS 3 Plugins - Plugin Development Part 3](https://gis-ops.com/qgis-3-plugin-tutorial-plugin-development-explained-part-3/)
+- [QGIS 3 Plugins - Set up Plugin Repository](https://gis-ops.com/qgis-3-plugin-tutorial-set-up-a-plugin-repository-explained/)
+
+
+---
+
+# QGIS 3 Plugins - Apply Best Practices & Useful Tips
```
Comment (TimMcCauley) [inline qgis/QGIS_PluginProperWay.md:16]: Generally we could use some titles that are a little more catchy... Geocoding with nominatim / opencage etc...? What do you think?

File: qgis/examples/quick_api_interactive_proper/quick_api/metadata.txt [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# This file contains metadata for your plugin.
+
+# This file should be included when you package your plugin.# Mandatory items:
+
+[general]
+name=Quick API
+qgisMinimumVersion=3.0
+description=Query OpenElevation API
```
Comment (TimMcCauley) [inline qgis/examples/quick_api_interactive_proper/quick_api/metadata.txt:8]: Misleading description, just noticed this when installing the plugin

**Review by TimMcCauley — APPROVED** [review]
Screenshot 2021-03-21 at 12 33 53

working well on OSX 10.15.6 and QGIS 3.10 - thanks @nilsnolde  !

File: qgis/QGIS_PluginProperWay.md (hunk 3 above)
Comment (nilsnolde) [reply qgis/QGIS_PluginProperWay.md:16]: Yep, sounds good. Then `QGIS 3 Plugins - Geocoding with Nominatim Part 1` - 3.

**Comment by nilsnolde** [comment]
Thanks!

BTW, GH has a new feature since a while: suggest changes. Cool thing is I can directly commit your suggestions in batch after you reviewed, really nice for typos etc.

File: qgis/QGIS_PluginProperWay.md (hunk 2 above)
Comment (nilsnolde) [reply qgis/QGIS_PluginProperWay.md:159]: yeah, I thought about using yapf (though not sure about compat with flake8)… Maybe it's time to give black a chance..

Comment (nilsnolde) [reply qgis/QGIS_PluginProperWay.md:159]: added

**Pull request merged**